### PR TITLE
Support finding all references.

### PR DESCRIPTION
**Pull request by rictic**
Apparently this code was written and tested, but never hooked up -.-

This change also completes the migration from the monolithic LocalEditorService!

 - [x] CHANGELOG.md has been updated

---
This change is [Reviewable](https://reviewable.io/reviews/polymer/polymer-editor-service/86)

### CODE REVIEW

**Comment by rictic** [comment]
Fixes https://github.com/Polymer/polymer-editor-service/issues/10

File: src/test/util.ts [hunk 1]
```diff
@@ -247,6 +247,17 @@ export class TestClient {
         CompletionRequest.type, params) as Promise<CompletionList>;
   }
 
+  async getReferences(
+      path: string, position: SourcePosition,
+      includeDefinition = false): Promise<Location[]> {
```
Comment (valdrinkoshi) [inline src/test/util.ts]: could you add a test that uses `includeDefinition = true`?

File: src/language-server/definition-finder.ts [hunk 2]
```diff
@@ -63,4 +67,46 @@ export default class DefinitionFinder extends Handler {
     }
     return feature.sourceRange;
   }
+
+  async getReferences(params: ReferenceParams): Promise<Location[]> {
```
Comment (valdrinkoshi) [inline src/language-server/definition-finder.ts]: docs pls 📖

File: src/language-server/definition-finder.ts [hunk 3]
```diff
@@ -63,4 +67,46 @@ export default class DefinitionFinder extends Handler {
...
+          locations.push(definition);
+        }
+      }
+    }
+    const localPath =
+        this.converter.getWorkspacePathToFile(params.textDocument);
+    const position = this.converter.convertPosition(params.position);
+    const analysis = await this.analyzer.analyzePackage();
+    const document = analysis.getDocument(localPath);
+    if (!(document instanceof Document)) {
```
Comment (valdrinkoshi) [inline src/language-server/definition-finder.ts]: when can this happen? e.g. if we loaded an img document?

File: src/language-server/definition-finder.ts [hunk 4]
```diff
@@ -63,4 +67,46 @@ export default class DefinitionFinder extends Handler {
...
+      return locations;
+    }
+    const location =
+        await this.featureFinder.getAstAtPosition(document, position);
+    if (!location) {
+      return locations;
```
Comment (valdrinkoshi) [inline src/language-server/definition-finder.ts]: when can this happen? e.g. empty document?

File: src/language-server/definition-finder.ts (hunk 3 above)
Comment (rictic) [reply src/language-server/definition-finder.ts]: Mostly if it had a parse error.

File: src/language-server/definition-finder.ts (hunk 4 above)
Comment (rictic) [reply src/language-server/definition-finder.ts]: Or if the position of the cursor is something that we don't have a location kind for, like a string literal, or some text.

File: src/test/util.ts (hunk 1 above)
Comment (rictic) [reply src/test/util.ts]: +1, done

File: src/language-server/definition-finder.ts (hunk 2 above)
Comment (rictic) [reply src/language-server/definition-finder.ts]: Done

**Comment by valdrinkoshi** [comment]
:shipit:

**Pull request merged**